src/CustomPromise.js

```diff
@@ -0,0 +1,27 @@
+export default class CustomPromise {
+}
+
+CustomPromise.all = (promises) => {
+  const results = [];
+
+  const merged = promises.reduce(
+    (accumulator, promise) =>
+      accumulator.
+      then(() => promise).
+      then(result => results.push(result)),
+    Promise.resolve(null));
+
+  return merged.then(() => results);
```

Comment:
It returns promise, since accumulator is set be default on Promise.resolve(null) which then gets chained with all the other promises. Added test to check it - CustomPromise all returns Promise